### PR TITLE
58307 update documents needed logic

### PR DESCRIPTION
**Pull request by jerekshoe**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
This mainly does 2 things:
* Update logic for determining if the `NeedFilesFromYou` component should be visible or not
* Strip out unwanted characters that we don't want to show, the main ones being: `<`, `>`, and `\n`

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#58307

## Testing done
Tests still pass
- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/claims-status/components/evss/ClaimStatusPageContent.jsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-  getCompletedDate,
-  itemsNeedingAttentionFromVet,
-} from '../../utils/helpers';
+import { getCompletedDate } from '../../utils/helpers';
 import ClaimComplete from '../ClaimComplete';
 import ClaimsDecision from '../ClaimsDecision';
 import ClaimsTimeline from '../ClaimsTimeline';
 import NeedFilesFromYou from '../NeedFilesFromYou';
 
+const itemsNeedingAttentionFromVet = events => {
```
Comment (jerekshoe) [inline src/applications/claims-status/components/evss/ClaimStatusPageContent.jsx:10]: Moving the EVSS-compatible version here

File: src/applications/claims-status/components/RequestedFilesInfo.jsx [hunk 2]
```diff
@@ -30,7 +26,7 @@ export default function RequestedFilesInfo({ id, filesNeeded, optionalFiles }) {
             <div className="item-container">
               <h3 className="file-request-title">{item.displayName}</h3>
               <p className="submission-description">
-                {truncateDescription(stripHtml(item.description))}
+                {truncateDescription(item.description)}
```
Comment (jerekshoe) [inline src/applications/claims-status/components/RequestedFilesInfo.jsx:29]: `truncateDescription` calls `stripHtml` internally now

File: src/applications/claims-status/utils/helpers.js [hunk 3]
```diff
@@ -241,28 +241,33 @@ export function getTopPosition(elem) {
...
   if (text && text.length > maxLength) {
     return `${text.substr(0, maxLength)}…`;
   }
-  return text;
-}
-
-// strip escaped html entities that have made its way into the desc
-export function stripHtml(text) {
-  return text && text.replace(/&\w+;/g, '');
+  return scrubDescription(text);
```
Comment (Mottie) [inline src/applications/claims-status/utils/helpers.js:262]: This seems like you should strip out the HTML before truncating the text, e.g. `<ThisIsMyReallyLongReactComponentNameButWillGetStrippedOut>Foo</ThisIsMyReallyLongReactComponentNameButWillGetStrippedOut>This important text may be truncated`.

Hmm, actually, I don't think the `stripHtml` function removes the text inside the `<>`

**Review by Mottie — APPROVED** [review]

File: src/applications/claims-status/utils/helpers.js (hunk 3 above)
Comment (jerekshoe) [reply src/applications/claims-status/utils/helpers.js:262]: It doesn't, what I'm usually seeing is thing like a description of `<VA Medical Facility>`. I want to keep the `VA Medical Facility` part, but want to remove the `<` and `>` characters. The EVSS version of the API returns this one as `&lt;VA Medical Facility&gt;` which is why this wasn't accounted for already

**Pull request merged**